### PR TITLE
Under the Hood view implementation blog post

### PR DESCRIPTION
**Pull request by davidjohnoliver**

### CODE REVIEW

File: doc/blog/201808-UnoUnderTheHoodButton.md [hunk 1]
```diff
@@ -0,0 +1,145 @@
+# How Uno implements views
```
Comment (Dissolute) [inline doc/blog/201808-UnoUnderTheHoodButton.md]: TODO: Catchier title

File: doc/blog/201808-UnoUnderTheHoodButton.md [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+
+2. Implement data-binding; 
+
+3. Implement the suite of views in the [UWP framework](https://docs.microsoft.com/en-us/windows/uwp/design/controls-and-patterns/controls-by-function) for each platform. 
+
+ 
+
+In this article I want to focus on that last point. How does Uno implement UWP's views? As a case study I'm going to use that humbly ubiquitous UI control, the button. 
+
+## Pushing the right buttons
```
Comment (MatFillion) [inline doc/blog/201808-UnoUnderTheHoodButton.md]: this could be the catchier title.

File: doc/blog/201808-UnoUnderTheHoodButton.md [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+*Visual tree information for UWP, Android, iOS, and WASM*
+
+So what does that get us? You can see the resulting visual trees on each target platform. They're all substantially similar. The top-level wrapping varies a little, but inside we can see the `MainPage`, `StackPanel`, `TextBlock` and `Button` we defined in XAML. (The `Frame` is created in the stock `App.xaml.cs` code that comes with the project template.)  
+
+You might notice there are a couple of extra views inside the `Button` that weren't explicitly defined in the XAML. These are part of Button's [default template](https://msdn.microsoft.com/en-us/library/windows/apps/mt299109.aspx?f=255&MSPPError=-2147217396). If you're not familiar with UWP/WPF/Silverlight/etc's concept of templating, there's a lot to say about [the subject](https://docs.microsoft.com/en-us/windows/uwp/design/controls-and-patterns/control-templates), but the gist of it is that any view which inherits from `Control` is a tabula rasa, an empty vessel which will be filled with subviews as defined in its template. (Some of these subviews might be templated controls themselves.)
```
Comment (jeromelaban) [inline doc/blog/201808-UnoUnderTheHoodButton.md]: Remove the error params from the default template link

Comment (jeromelaban) [reply doc/blog/201808-UnoUnderTheHoodButton.md]: >  will be filled with subviews 

_inner elements_ instead ? or _inner UIElements_ ? subviews is probably too UIKit specific.

File: doc/blog/201808-UnoUnderTheHoodButton.md [hunk 4]
```diff
@@ -0,0 +1,145 @@
...
+
+In WebAssembly, for now the inheritance hierarchy is a bit simpler and `UIElement` is at the root of the type tree. As you can see in the screenshot, Uno.WASM is generating `<div>` elements for each view in the visual tree. 
+
+## Style points
+
+The code above nets us a very plain, workaday button, but we could easily spice it up. We set the `Content` property to a text string, but `Content` can be anything, even another view. Our button could be an image, a shape, or a complex visual hierarchy. It could even have another button inside of it.
```
Comment (jeromelaban) [inline doc/blog/201808-UnoUnderTheHoodButton.md]: We could link to Playground snippet samples ?

File: doc/blog/201808-UnoUnderTheHoodButton.md (hunk 2 above)
Comment (Dissolute) [reply doc/blog/201808-UnoUnderTheHoodButton.md]: Sure :)

File: doc/blog/201808-UnoUnderTheHoodButton.md (hunk 3 above)
Comment (Dissolute) [reply doc/blog/201808-UnoUnderTheHoodButton.md]: Good point

**Pull request merged**